system/src/system_network_compat.cpp

```diff
@@ -40,6 +40,7 @@ uint32_t wlan_watchdog_duration = 0;
 
 volatile uint8_t SPARK_WLAN_RESET = 0;
 volatile uint8_t SPARK_WLAN_SLEEP = 0;
+volatile uint8_t SPARK_WLAN_CONNECT_RESTORE = 0;
```

Comment:
How about `SPARK_WLAN_CONNECT` instead? Can be set in `network_connect()` and where appropriate as well (e.g. `manage_network_connection` or restored after sleep).

Reply:
I was considering to use `SPARK_WLAN_CONNECT`, yet we need to set/clear it in many places to make sure its state is well set, that will even complicate the system network manager for Gen2 and easier to cause race condition under unknown conditions. So I just prefix it with `_RESTORE` so that it just need to be set where necessary, like resetting the modem or after sleep. The caller who has called `network_off()` should take the responsibility to see if the connection should be restored.